### PR TITLE
feat: add prom metrics endpoint generated from postgres data

### PR DESCRIPTION
**Pull request by alanshaw**
Not exporting byte totals anymore as these can come from dagcargo.

Have split upload totals out by type.

Have split pin totals by service and status (as opposed to just status).

Example output:

```
# HELP nftstorage_users_total Total users registered.
# TYPE nftstorage_users_total counter
nftstorage_users_total 1
# HELP nftstorage_uploads_total Total number of Car uploads.
# TYPE nftstorage_uploads_total counter
nftstorage_uploads_total{type:"Car"} 0
# HELP nftstorage_uploads_total Total number of Blob uploads.
# TYPE nftstorage_uploads_total counter
nftstorage_uploads_total{type:"Blob"} 0
# HELP nftstorage_uploads_total Total number of Multipart uploads.
# TYPE nftstorage_uploads_total counter
nftstorage_uploads_total{type:"Multipart"} 0
# HELP nftstorage_uploads_total Total number of Remote uploads.
# TYPE nftstorage_uploads_total counter
nftstorage_uploads_total{type:"Remote"} 0
# HELP nftstorage_uploads_total Total number of Nft uploads.
# TYPE nftstorage_uploads_total counter
nftstorage_uploads_total{type:"Nft"} 0
# HELP nftstorage_pins_total Total number of Pinata pins that are PinQueued.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"Pinata",status:"PinQueued"} 0
# HELP nftstorage_pins_total Total number of Pinata pins that are Pinning.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"Pinata",status:"Pinning"} 0
# HELP nftstorage_pins_total Total number of Pinata pins that are Pinned.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"Pinata",status:"Pinned"} 0
# HELP nftstorage_pins_total Total number of Pinata pins that are PinError.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"Pinata",status:"PinError"} 0
# HELP nftstorage_pins_total Total number of IpfsCluster pins that are PinQueued.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"IpfsCluster",status:"PinQueued"} 0
# HELP nftstorage_pins_total Total number of IpfsCluster pins that are Pinning.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"IpfsCluster",status:"Pinning"} 0
# HELP nftstorage_pins_total Total number of IpfsCluster pins that are Pinned.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"IpfsCluster",status:"Pinned"} 0
# HELP nftstorage_pins_total Total number of IpfsCluster pins that are PinError.
# TYPE nftstorage_pins_total counter
nftstorage_pins_total{service:"IpfsCluster",status:"PinError"} 0
```

### CODE REVIEW

File: packages/api/src/routes-v1/metrics.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { secrets, database } from '../constants.js'
+import { DBClient } from '../utils/db-client.js'
+
+const db = new DBClient(database.url, secrets.database)
+
+export async function getUserMetrics() {
+  const query = db.client.from('account')
```
Comment (hugomrdias) [inline packages/api/src/routes-v1/metrics.js:7]: this needs to be updated to `user`

File: packages/api/src/routes-v1/metrics.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const res = await query.select('*', { head: true, count: 'exact' })
```
Comment (hugomrdias) [inline packages/api/src/routes-v1/metrics.js:8]: do we need the count to be exact ? maybe we should measure how long does the other modes take with a full db

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: packages/api/src/routes-v1/metrics.js (hunk 2 above)
Comment (alanshaw) [reply packages/api/src/routes-v1/metrics.js:8]: I think for users it is fine for now but I was thinking the same thing for uploads. Lets see how long it takes and we can switch to fuzzy count if problematic.

File: packages/api/src/routes-v1/metrics.js (hunk 1 above)
Comment (alanshaw) [reply packages/api/src/routes-v1/metrics.js:7]: Good catch!

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**